.github/workflows/ci-pipeline.yml

```diff
@@ -18,3 +18,42 @@ jobs:
     - run: docker build --target test --tag test .
     - run: docker run test todo_app/test/unit_tests
     - run: docker run --env-file .env.docker test todo_app/test/integration_tests
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
```

Comment:
We _probably_ don't want to version this so specifically to a particular commit - doing so leaves us prone to missing future patches if any concerns are identified